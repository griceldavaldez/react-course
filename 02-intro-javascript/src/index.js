console.log('Intro a Javascript');