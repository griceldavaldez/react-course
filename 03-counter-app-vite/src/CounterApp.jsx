import { useState } from 'react';
import PropTypes from 'prop-types';


//Cuando hay un cambio en el estado se va a volver a disparar el functional component
export function CounterApp({value}){
    
    const [counter, setCounter] = useState(value);

    const handleAddition = () =>{
        setCounter(counter + 1);
        //setCounter((c) => c+1)
    }

    const handleSubtract = () =>{
        setCounter(counter - 1);
    }

    const handleReset = () =>{
        setCounter(value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>  
            <button aria-label='btn-add' onClick={handleAddition}> +1 </button> 
            <button aria-label='btn-subt' onClick={handleSubtract}> -1 </button> 
            <button aria-label='btn-reset' onClick={handleReset}> Reset </button> 
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}


CounterApp.defaultProps = {
    value: 123
}