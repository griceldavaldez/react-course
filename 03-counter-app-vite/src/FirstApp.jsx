//import { Fragment } from "react";

/*export function FirstApp(){
    return (
        <Fragment>
            <h1>Gricelda</h1>
            <p>Soy un subtitulo</p>
        </Fragment>
    );
}*/


//Un fragmento es un agrupador de otros elementos html
/*export function FirstApp(){
    return (
        <>
            <h1>Gricelda</h1>
            <p>Soy un subtitulo</p>
        </>
    );
}*/

// <></>  otra forma de nombrar fragmentos

/*const name = 'Gricelda';
const boolean = true;
const arreglo = [1,2,3,4,5,6,7,8];
const objeto = {
    nombre: 'App',
    value: 5
}

const multiplicacion = (a,b)=>{
    return a*b;
}


export function FirstApp(props){
    console.log(props);
    return (
        <>
            <h1>{props.title}</h1>
            <h1>{name}</h1>
            <h1>{boolean}</h1>
            <h1>{arreglo}</h1>
            <code>{JSON.stringify(objeto) //si quiero imprimir objeto, puedo hacerlo con esta funcion}</code>
            <p>{1+2}</p>
            <p>{multiplicacion(10,2)}</p>
        </>
    );
}*/

//{} sirve para agregar codigo JS en html
//sirve solo para expresiones o valores, no para objetos
//si quiero usar objetos se usa JSON.stringify(objeto)


/*PropTypes*/

//para definir el tipo de dato de los props
import PropTypes from 'prop-types';

export function FirstApp({title, subtitle=123, name}){
    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <h1>{subtitle}</h1> 
            <h1>{name}</h1>        
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}


FirstApp.defaultProps = {
    //title: 'No hay titulo',
    subtitle: "No hay subtitulo",
    name: 'Gricelda'
}