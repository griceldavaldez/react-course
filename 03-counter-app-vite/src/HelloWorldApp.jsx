export function App(){
    //document.createElement...
    return <h1>Hola mundo!!!</h1>;
}

export const print = () => {
    console.log('Hola mundo');
}

/*
function App(){
    //document.createElement...
    return <h1>Hola mundo!!!</h1>;
}

export default App;
*/

/*import React from 'react'

function HelloWorldApp() {
  return (
    <div>HelloWorldApp</div>
  )
}

export default HelloWorldApp*/