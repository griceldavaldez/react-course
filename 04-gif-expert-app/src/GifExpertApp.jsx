import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
    //Los hooks states se manejan de manera posicional
    //No se deben declarar hooks states de manera condicional
    const [categories, setCategories] = useState(['Barbie']);

    console.log('setCategories', setCategories);
    const onAddCategory = (newCategory) =>{
        //if(categories.includes(newCategory)) return;
        if((categories.map((category) => category.toLowerCase())).includes(newCategory.toLowerCase())) return;
        setCategories([newCategory,...categories]);
    }

    return (
    <>
        <h1>GifExpertApp</h1>

        <AddCategory 
            //setCategories={setCategories}
            onNewCategories={(event) => onAddCategory(event)}
        />

        {
            categories.map((category) => (
                <GifGrid 
                    key={category} 
                    category={category}
                />       
            ))
        }

    </>
  )
}
