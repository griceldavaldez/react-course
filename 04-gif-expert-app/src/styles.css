* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: rgb(234, 234, 234);
  }
  
  body {
    padding: 60px;
  }
  
  input {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgb(97, 32, 158);
    color: black;
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 100%;
  }
  
  h1 {
    font-size: 3.5rem;
    color: blueviolet;
  }

  h2 {
    font-size: 1.5rem;
  }
  
  h3 {
    font-size: 3rem;
    margin-bottom: 5px;
  }
  
  /*.card-grid {
    display: flex;
    flex: 3;
    flex-direction: row;
    flex-wrap: wrap;
    /*justify-content: center;*/
    /*width: calc(33.33% - 20px);*/
 /* }*/
  
  /*.card {
    align-content: center;
    align-items: center;
    background-color: white;
    border-radius: 10px;
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
    /*display: flex;*/
    /*flex: 3;
    flex-direction: column;
    height: 0%;  
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
  }*/

  .card-grid {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    margin: -20px;
}

.card {
    flex-basis: calc(33.33% - 40px);
    margin: 20px;
    background-color: white;
    border-radius: 10px;
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;

    height: auto; /* O ajusta a un valor fijo si es necesario */
    display: flex;
    flex-direction: column;
    align-items: center;

}

  
  /*.card p {
    background-color: white;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
  }*/

  .card p {
    background-color: white;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
}
  
  /*.card img {
    width: 100%;
  }*/

  .card img {
    width: 410px;
    height: 410px;
    object-fit: cover;
}


.alert-danger {
  background-color: #f8d7da;
  border: 1px solid #f5c6cb;
  color: #721c24;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border-radius: 0.25rem;
}

.alert-danger strong {
  font-weight: bold;
  background-color: #f8d7da;
}
