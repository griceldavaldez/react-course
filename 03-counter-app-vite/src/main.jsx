import React from 'react'
import ReactDOM from 'react-dom/client'

//si exporto individualmente(es con desestructuracion y utilizando nombre original)
//import {App, print} from './HelloWorldApp';
import { FirstApp } from './FirstApp';

//si exporto como default(puedo ponerle el nombre que quiera)
//import NameApp from ./HelloWorld 

import './styles.css';
import { CounterApp } from './CounterApp';

//se recomienda usar el nodo estricto
/*ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>  
        <FirstApp/>
    </React.StrictMode>
);*/

/*Estructura basica:
1. import de React y otras herramientas de renderizacion
2. functional component
3. renderizar el componente en el DOM, view, etc
*/


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>  
        {/*<CounterApp value={0}/>*/}
        <FirstApp title="Hola soy Vegeta"/>
    </React.StrictMode>
);