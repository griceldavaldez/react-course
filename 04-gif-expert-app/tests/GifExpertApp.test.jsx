import { render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';



describe('Pruebas en <GifExpertApp />', () => {


    test('debe renderizar GifExpertApp sin errores', () => {
        const {container} = render(<GifExpertApp />);
        expect( container ).toMatchSnapshot();
    });

    test('debe agregar una nueva categoría al enviar el formulario', () => {

        //TODO no funciona la prueba. Arreglar

      /*  const { getByLabelText, getByText } = render(<GifExpertApp />);

        const input = getByLabelText("Category");
        fireEvent.change(input, { target: { value: "One Punch" } });

        const addButton = getByText("Add");
        fireEvent.click(addButton);

        // Verificar si la nueva categoría se ha agregado correctamente
        expect(getByText("One Punch")).toBeInTheDocument();*/
        
    });


    test('no debe agregar una categoría duplicada', () => {

        //TODO no funciona la prueba. Arreglar

        /*const { getByLabelText, getByText } = render(<GifExpertApp />);

        const input = getByLabelText("Category");
        fireEvent.change(input, { target: { value: "Barbie" } });
      
        const addButton = getByText("Add");
        fireEvent.click(addButton);
      
        // Verificar que la categoría duplicada no se agregó
        expect(getByText("Barbie")).toBeInTheDocument();
        expect(getByText("Barbie").parentElement.childElementCount).toBe(1);*/

    });

    
});