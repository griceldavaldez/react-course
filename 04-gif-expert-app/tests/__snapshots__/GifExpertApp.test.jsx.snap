// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe renderizar GifExpertApp sin errores 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Barbie
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
